baccmod/spatial_fit_acceptance_map_creator.py

```diff
@@ -0,0 +1,207 @@
+        interpolation_cleaning_energy_relative_threshold: float, optional
+            To be considered value, the bin in energy need at least one adjacent bin with a relative difference within this range
+        interpolation_cleaning_spatial_relative_threshold: float, optional
+            To be considered value, the bin in space need at least one adjacent bin with a relative difference within this range
+        name_normalisation_parameter: string, optional
+            All the parameters containing this string in the model will be automatically normalised based on overall counts at the start of the fit
+        model_to_fit: Fittable2DModel, optional
+            The model to fit to the data
+        """
+        super().__init__(
+            energy_axis=energy_axis,
+            offset_axis=offset_axis,
+            oversample_map=oversample_map,
+            exclude_regions=exclude_regions,
+            cos_zenith_binning_method=cos_zenith_binning_method,
+            cos_zenith_binning_parameter_value=cos_zenith_binning_parameter_value,
+            initial_cos_zenith_binning=initial_cos_zenith_binning,
+            max_angular_separation_wobble=max_angular_separation_wobble,
+            zenith_binning_run_splitting=zenith_binning_run_splitting,
+            max_fraction_pixel_rotation_fov=max_fraction_pixel_rotation_fov,
+            time_resolution=time_resolution,
+            use_mini_irf_computation=use_mini_irf_computation,
+            mini_irf_time_resolution=mini_irf_time_resolution,
+            interpolation_type=interpolation_type,
+            activate_interpolation_cleaning=activate_interpolation_cleaning,
+            interpolation_cleaning_energy_relative_threshold=interpolation_cleaning_energy_relative_threshold,
+            interpolation_cleaning_spatial_relative_threshold=interpolation_cleaning_spatial_relative_threshold,
+            name_normalisation_parameter=name_normalisation_parameter
+        )
+
+        self.model_to_fit = model_to_fit
+
+    def create_acceptance_map(self, observations) -> Background3D:
+        """
+        Calculate a 3D grid acceptance map by fitting 2D model in each energy slice
+
+        Parameters
+        ----------
+        observations : gammapy.data.observations.Observations
+            The collection of observations used to make the acceptance map
+
+        Returns
+        -------
+        acceptance_map : gammapy.irf.background.Background3D
+        """
+
+        # 1) gather fine-binned counts + exposures + livetime
+        count_background, exp_map_background, exp_map_background_total, livetime = self._create_base_computation_map(
+            observations
+        )
+
+        # 2) downsample exposures
+        exp_ds = exp_map_background.downsample(self.oversample_map, preserve_counts=True)
+        exp_total_ds = exp_map_background_total.downsample(self.oversample_map, preserve_counts=True)
```

Comment:
Should we consider fitting on the fine maps?

Reply:
It's a possibility, it was not what was done with the current implementation. I guess the main drawback will be compute time.

Reply:
It may also be less stable. Maybe something to investigate later.